server/rest/src/main/java/org/infinispan/rest/server/InfinispanResponse.java

```diff
@@ -0,0 +1,153 @@
+package org.infinispan.rest.server;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.rest.server.operations.mediatypes.MediaType;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http2.HttpConversionUtil;
+
+public class InfinispanResponse {
```

Comment:
> Yes, we need it.

Why? Do we plan to switch away from Netty soon?

> I'm a big fan of Optionals.

I think it's clear that I'm not a big fan of `Optional` :) To me, you're just moving complexity away from where you're using the value to where you're creating the `Optional`, so you're not really gaining anything. E.g. instead of having `Optional` fields and a separate method that transforms those fields into headers, the setters could set add the headers to the response directly.

>  I bet OpenJDK engineers implemented a lot of optimizations since then.

`-XX:MaxBCEAEstimateSize` still defaults to 150 in the latest JDK8 and JDK9 builds.